### PR TITLE
Remove alert for pre-existing new columns while merging realtime schema

### PR DESCRIPTION
**Pull request by findingrish**
Currently, an alert is thrown while merging datasource schema with realtime segment schema when the datasource schema already has update columns from the delta schema. 

This isn't an error condition since the datasource schema can have those columns from a different segment. 

One scenario in which this can occur is when multiple replicas for a task is run. 

This PR has:

- [X] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: server/src/main/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCache.java [hunk 1]
```diff
@@ -826,22 +824,20 @@ Optional<RowSignature> mergeOrCreateRowSignature(
...
       }
 
-      if (missingUpdateColumns || existingNewColumns) {
+      if (!missingUpdateColumns.isEmpty()) {
         log.makeAlert(
-            "Error merging delta schema update with existing row signature. segmentId [%s], "
-            + "existingSignature [%s], deltaSchema [%s], missingUpdateColumns [%s], existingNewColumns [%s].",
-            segmentId,
+            "Realtime schema merge: datasource [%s] is missing columns to be updated. "
+            + "ExistingSignature [%s], deltaSchema [%s], missingUpdateColumns [%s].",
```
Comment (kfaraz) [inline server/src/main/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCache.java:836]: The alert description should be small. Don't include big objects like the `existingSignature`or `deltaSchema` in it. You may add these fields separately using `.addData()`.

File: server/src/main/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCache.java [hunk 2]
```diff
@@ -826,23 +824,19 @@ Optional<RowSignature> mergeOrCreateRowSignature(
...
-            "Error merging delta schema update with existing row signature. segmentId [%s], "
-            + "existingSignature [%s], deltaSchema [%s], missingUpdateColumns [%s], existingNewColumns [%s].",
-            segmentId,
-            existingSignature,
-            segmentSchema,
-            missingUpdateColumns,
-            existingNewColumns
-        ).emit();
+      if (!missingUpdateColumns.isEmpty()) {
+        log.makeAlert("Datasource schema is missing columns to be updated in delta realtime segment schema.")
```
Comment (kfaraz) [inline server/src/main/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCache.java:834]: The message is confusing. Could you try to rephrase this to say exactly what went wrong and what can be done to remediate?

File: server/src/main/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCache.java [hunk 3]
```diff
@@ -811,13 +811,11 @@ Optional<RowSignature> mergeOrCreateRowSignature(
       Map<String, ColumnType> columnMapping = segmentSchema.getColumnTypeMap();
 
       // column type to be updated is not present in the existing schema
-      boolean missingUpdateColumns = false;
-      // new column to be added is already present in the existing schema
-      boolean existingNewColumns = false;
+      Set<String> missingUpdateColumns = new HashSet<>();
```
Comment (kfaraz) [inline server/src/main/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCache.java:814]: Nit:
```suggestion
      final Set<String> missingUpdateColumns = new HashSet<>();
```

**Review by cryptoe — APPROVED** [review]

**Review by kfaraz — APPROVED** [review]
